libs/community/langchain_community/vectorstores/cloudSQL.py

```diff
@@ -0,0 +1,330 @@
+from __future__ import annotations
+
+import uuid
+
+from typing import Any, Callable, Dict, List, Optional, Tuple, Type
+
+from langchain_core.documents import Document
+from langchain_core.vectorstores import VectorStore
+from langchain_core.embeddings import Embeddings
+
+from google.cloud.sql.connector import Connector
+import google.auth
+
+import sqlalchemy
+from sqlalchemy import inspect
+from sqlalchemy import Table, Column, Integer, String
+from sqlalchemy.dialects.postgresql import JSON, UUID
+
+from pgvector.sqlalchemy import Vector
+
+def _get_IAM_user(credentials):
+"""Get user/service account name"""
+import google.auth
+from google.auth.transport.requests import Request
+from google.auth.compute_engine import _metadata
+
+# Checks for service account
+if hasattr(credentials, "service_account_email"):
+    return credentials.service_account_email[:-20]
+else:
+    # Defaults to getting the user account
+    command = ["gcloud", "auth", "list", "--filter=status:ACTIVE", "--format=value(account)"]
+    result = subprocess.run(command, stdout=subprocess.PIPE, text=True)
+    if result.returncode == 0:
+        return result.stdout.strip()
+    else:
+        return f"Error: {result.stderr}"
+        
+class CloudSQLEngine:
+    """Creating a connection to the CloudSQL instance
+    To use, you need the following packages installed:
+        cloud-sql-python-connector[asyncpg]
+    """
+    
+    connector = Connector()
+    
+    @staticmethod
+    def from_instance(
+        region: str, 
+        instance: str, 
+        database: str, 
+        project_id: str=None
+    ) -> sqlalchemy.engine.Engine:
```

Comment:
This should return an instance of CloudSQLEngine

Reply:
Should we use CloudSQLPostgresEngine instead? Since we will have three packages for 3 databases, it's better to distinguish the engine type by name.